internal/resources/private_endpoint_service.go

```diff
@@ -0,0 +1,360 @@
+func (p *PrivateEndpointService) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	data, ok := req.ProviderData.(*providerschema.Data)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *providerschema.Data, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	p.Data = data
+}
+
+// ImportState imports a private endpoint service status.
+func (p *PrivateEndpointService) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	resource.ImportStatePassthroughID(ctx, path.Root("cluster_id"), req, resp)
+}
+
+// validateCreateEndpointService ensures organization id, project id and cluster id are valued.
+func validateCreateEndpointService(plan providerschema.PrivateEndpointService) error {
+	if plan.OrganizationId.IsNull() {
+		return errors.ErrOrganizationIdCannotBeEmpty
+	}
+	if plan.ProjectId.IsNull() {
+		return errors.ErrProjectIdMissing
+	}
+	if plan.ClusterId.IsNull() {
+		return errors.ErrClusterIdMissing
+	}
+
+	return nil
+}
+
+// initializePlan initializes an instance of providerschema.PrivateEndpointService
+// with the specified plan. It marks all computed fields as null.
+func initializePrivateEndpointServicePlan(plan providerschema.PrivateEndpointService) providerschema.PrivateEndpointService {
+	if plan.Enabled.IsNull() || plan.Enabled.IsUnknown() {
+		plan.Enabled = types.BoolNull()
+	}
+	return plan
+}
+
+// waitUntilStatusChanges terraform will wait until the service status changes on the cluster.
+func (p *PrivateEndpointService) waitUntilStatusChanges(ctx context.Context, finalState bool, organizationId, projectId, clusterId string) error {
+	var cancel context.CancelFunc
+	ctx, cancel = context.WithTimeout(ctx, time.Minute*60)
+	defer cancel()
+
+	timer := time.NewTimer(time.Minute * 1)
+
+	for {
+		select {
+		case <-ctx.Done():
+			const msg = "changing private endpoint service status timed out after initiation"
+			return fmt.Errorf(msg)
+
+		case <-timer.C:
+			response, err := p.getServiceStatus(ctx, organizationId, projectId, clusterId)
+			if err != nil {
+				return err
+			}
+
+			if response.Enabled == finalState {
+				return nil
+			}
+			timer.Reset(time.Minute * 1)
+		}
+	}
+}
+
+// getServiceStatus retrieves current private endpoint service status.
+func (p *PrivateEndpointService) getServiceStatus(ctx context.Context, organizationId, projectId, clusterId string) (*api.GetPrivateEndpointServiceStatusResponse, error) {
+	url := fmt.Sprintf("%s/v4/organizations/%s/projects/%s/clusters/%s/privateEndpointService", p.HostURL, organizationId, projectId, clusterId)
+	cfg := api.EndpointCfg{Url: url, Method: http.MethodGet, SuccessStatus: http.StatusOK}
+	response, err := p.Client.ExecuteWithRetry(
+		ctx,
+		cfg,
+		nil,
+		p.Token,
+		nil,
+	)
+	if err != nil {
```

Comment:
We need to handle the case when 404 is being returned, as it should not come to user as error, it should come as resource has been changed outside of terraform provider.

Reply:
Addressed in 678457985aba6523712b139fa613086a83c825af